src/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.netcoreapp.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public abstract partial class FileSystemWatcherTest : FileCleanupTestBase
+    {
+        /// <summary>
+        /// Does verification that the given watcher will throw exactly/only the events in "expectedEvents" when
+        /// "action" is executed.
+        /// </summary>
+        /// <param name="watcher">The FileSystemWatcher to test</param>
+        /// <param name="expectedEvents">All of the events that are expected to be raised by this action</param>
+        /// <param name="action">The Action that will trigger events.</param>
+        /// <param name="cleanup">Optional. Undoes the action and cleans up the watcher so the test may be run again if necessary.</param>
+        /// <param name="expectedPath">Optional. Adds path verification to all expected events.</param>
+        /// <param name="attempts">Optional. Number of times the test should be executed if it's failing.</param>
+        public static void ExpectEvent(FileSystemWatcher watcher, WatcherChangeTypes expectedEvents, Action action, Action cleanup = null, string[] expectedPaths = null, int attempts = DefaultAttemptsForExpectedEvent, int timeout = WaitForExpectedEventTimeout)
+        {
+            int attemptsCompleted = 0;
+            bool result = false;
+            FileSystemWatcher newWatcher = watcher;
+            while (!result && attemptsCompleted++ < attempts)
+            {
+                if (attemptsCompleted > 1)
+                {
+                    // Re-create the watcher to get a clean iteration.
+                    newWatcher = new FileSystemWatcher()
```

Comment:
Don't duplicate the entire method. Add a new helper that does only what is different (i.e. `FileSystemWatcher RecreateWatcher(FileSystemWatcher watcher)` or something like that).